sequence_processing_pipeline/Pipeline.py

```diff
@@ -238,12 +254,21 @@ def generate_sample_information_files(self):
         Generate sample-information files in self.output_path.
         :return: A list of paths to sample-information-files.
         """
-        samples = []
-        for sample in self.sample_sheet.samples:
-            if sample['Sample_ID'].startswith('BLANK'):
-                samples.append((sample['Sample_ID'], sample['Sample_Project']))
+        if self.mapping_file is not None:
+            # TODO: Note we may need to introduce a conversion to sample_id
+            #  format from sample_name.
+            df = self.mapping_file[['sample_name', 'project_name']]
+            samples = list(df.to_records(index=False))
+            samples = [x for x in samples if 'BLANK' in x[0]]
```

Comment:
Why is this `'BLANK' in x[0]` vs. `x[0].startswith('BLANK')`?

Reply:
From the previous comment: I left this comment for discussion with you and MacKenzie. The BLANKS in the new mapping file are of the form '11052.BLANK242.11E'. Assuming that's correct form for sample-sheets, we can't assume that BLANKs start with 'BLANK'.